File: medium/python/334.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        smallest = nums[0]
        second_smallest = None

        for num in nums:
            if num < smallest:
                smallest = num
            elif num > smallest:
                if second_smallest is None or num < second_smallest:
                    second_smallest = num
                elif num > second_smallest:
                    return True
        return False

File: medium/ts/334.ts
// https://leetcode.com/problems/increasing-triplet-subsequence/?envType=study-plan-v2&envId=leetcode-75
// Time: O(n)
// Space: O(1)
function increasingTriplet(nums: number[]): boolean {
  let smallest = nums[0];
  let secondSmallest;

  for (let i = 0; i < nums.length; i++) {
    let num = nums[i];
    if (num < smallest) {
      smallest = num;
    } else if (num > smallest) {
      if (secondSmallest === undefined || num < secondSmallest) {
        secondSmallest = num;
      } else if (num > secondSmallest) {
        return true;
      }
    } else if (num > smallest && num > secondSmallest) {
      return true;
    }
  }
  return false;
}
